src/Controller/ArticleController.php

```diff
@@ -172,4 +177,40 @@ public function changeIsPublished(Article $article): Response
 
         return $this->redirectToRoute('article_list');
     }
+
+    /**
+     * @Route("/download", name="download")
+     */
+    public function downloadPdf(Request $request, \Knp\Snappy\Pdf $snappy)
+    {
+        $filename = 'temp/'.md5(uniqid()).'.pdf';
+        $article = $this->getDoctrine()->getManager()->find(Article::class, $request->get('id'));
+
+        $snappy->generateFromHtml(
+            $this->renderView(
+                'articles/show.html.twig',
+                ['article' => $article],
+            ),
+            $filename
+        );
+
+        $response = new BinaryFileResponse(new File($filename));
+        $response->deleteFileAfterSend(true);
+        $response->headers->set('Content-Type', 'application/pdf');
+        $response->setContentDisposition(ResponseHeaderBag::DISPOSITION_ATTACHMENT, 'article.pdf');
```

Comment:
jak pobiorę 3 różne artykuły to będę miał na komputerze **article.pdf**, **article (1).pdf**, **article (2).pdf**. Oczywiście mogę sam to zmienić, ponieważ komputer pyta nas przed zapisaniem o nazwę, ale tak nie powinno być. Tutaj powinieneś zwrócić plik o unikatowej nazwie, np. tytuł artykułu, a jeszcze lepiej jakbyś uwzględnił jakieś unikatowe pole w encji Article.